modules/dr-restoring-cluster-state.adoc

```diff
@@ -0,0 +1,77 @@
+
+.Prerequisites
+
+* SSH access to master hosts.
+* A backed up etcd snapshot.
+
+.Procedure
+
+Perform the following steps on each master host in your cluster.
+
+. Copy the etcd snapshot file to a master host.
++
+This procedure assumes that you have copied a snapshot file called `snapshot.db` to the `/home/core/` directory of your master host.
+
+. Access the master host.
+
+. Set the `INITIAL_CLUSTER` variable to the list of members in the format of `<name>=<url>`. This variable will be passed to the script and must be exactly the same for each member.
++
+----
+[core@ip-10-0-143-125 ~]$ export INITIAL_CLUSTER="etcd-member-ip-10-0-143-125.ec2.internal=etcd-0.clustername.devcluster.openshift.com:2380,etcd-member-ip-10-0-35-108.ec2.internal=etcd-1.clustername.devcluster.openshift.com:2380,etcd-member-ip-10-0-10-16.ec2.internal=etcd-2.clustername.devcluster.openshift.com:2380"
+----
+
+. Run the `etcd-snapshot-restore.sh` script. Pass in two parameters: the path to the snapshot file and list of members, which is defined by the `INITIAL_CLUSTER` variable.
++
+----
+[core@ip-10-0-143-125 ~]$ sudo /usr/local/bin/etcd-snapshot-restore.sh /home/core/snapshot.db $INITIAL_CLUSTER
+Creating asset directory ./assets
+Downloading etcdctl binary..
+etcdctl version: 3.3.10
+API version: 3.3
+Backing up /etc/kubernetes/manifests/etcd-member.yaml to ./assets/backup/
+Stopping all static pods..
+..stopping kube-scheduler-pod.yaml
+..stopping kube-controller-manager-pod.yaml
+..stopping kube-apiserver-pod.yaml
+..stopping etcd-member.yaml
+Stopping etcd..
+Waiting for etcd-member to stop
+Stopping kubelet..
+Stopping all containers..
+bd44e4bc942276eb1a6d4b48ecd9f5fe95570f54aa9c6b16939fa2d9b679e1ea
+d88defb9da5ae623592b81619e3690faeb4fa645440e71c029812cb960ff586f
+3920ced20723064a379739c4a586f909497a7b6705a5b3cf367d9b930f23a5f1
+d470f7a2d962c90f3a21bcc021970bde96bc8908f317ec70f1c21720b322c25c
+Backing up etcd data-dir..
+Removing etcd data-dir /var/lib/etcd
+Restoring etcd member etcd-member-ip-10-0-143-125.ec2.internal from snapshot..
+2019-05-15 19:03:34.647589 I | pkg/netutil: resolving etcd-0.clustername.devcluster.openshift.com:2380 to 10.0.143.125:2380
+2019-05-15 19:03:34.883545 I | mvcc: restore compact to 361491
+2019-05-15 19:03:34.915679 I | etcdserver/membership: added member cbe982c74cbb42f [https://etcd-0.clustername.devcluster.openshift.com:2380] to cluster 807ae3bffc8d69ca
+Starting static pods..
+..starting kube-scheduler-pod.yaml
+..starting kube-controller-manager-pod.yaml
+..starting kube-apiserver-pod.yaml
+..starting etcd-member.yaml
+Starting kubelet..
+----
+
+. Wait for the API to come back up.
+
+. Verify that the Machine Configs have been applied using the following command.
++
+----
+$ oc get machineconfigpool
+NAME     CONFIG                                             UPDATED   UPDATING
+master   rendered-master-50e7e00374e80b767fcc922bdfbc522b   True      False
+----
+
+. Repeat these steps on your other master hosts.
```

Comment:
`Repeat these steps...` should be moved earlier, to before `Wait for the API to come back up`.